packages/ui-components/src/components/DateTimePicker/DateTimePicker.component.tsx

```diff
@@ -0,0 +1,651 @@
+    setIsOpen(false)
+    onClose && onClose(selectedDate, dateStr)
+  }
+
+  const handleMonthChange = (selectedDate: Date[], dateStr: string, _instance: flatpickr.Instance) => {
+    setTheDate({ selectedDate: selectedDate, selectedDateStr: dateStr })
+    onMonthChange && onMonthChange(selectedDate, dateStr)
+  }
+
+  const handleOpen = (selectedDate: Date[], dateStr: string, _instance: flatpickr.Instance) => {
+    setIsOpen(true)
+    onOpen && onOpen(selectedDate, dateStr)
+  }
+
+  const handleReady = (selectedDate: Date[], dateStr: string, _instance: flatpickr.Instance) => {
+    onReady && onReady(selectedDate, dateStr)
+  }
+
+  const handleYearChange = (selectedDate: Date[], dateStr: string, _instance: flatpickr.Instance) => {
+    setTheDate({ selectedDate: selectedDate, selectedDateStr: dateStr })
+    onYearChange && onYearChange(selectedDate, dateStr)
+  }
+
+  const handleInputFocus = () => {
+    setHasFocus(true)
+    onFocus && onFocus(theDate.selectedDate, theDate.selectedDateStr)
+  }
+
+  const handleClearIconClick = () => {
+    const instance = flatpickrInstanceRef.current
+    setTheDate({})
+    instance?.clear()
+    onClear && onClear([], "")
+  }
+
+  // Create stringified versions of the value prop and its aliases in order to use them in a useEffect dependency array later.
+  const stringifiedValue = JSON.stringify(value)
+  const stringifiedDefaultDate = JSON.stringify(defaultDate)
+  const stringifiedDefaultValue = JSON.stringify(defaultValue)
+
+  // Function to determine the date format. Will return the dateFormat if passed as a prop, or a useful defaultFormat depending on whether the DateTimePicker is set to show the time, seconds, or no calendar at all (time picker only).
+  const getDateFormat = () => {
+    const defaultDateFormat = enableTime
+      ? noCalendar
+        ? enableSeconds
+          ? "H:i:S"
+          : "H:i"
+        : enableSeconds
+          ? "Y-m-d H:i:S"
+          : "Y-m-d H:i"
+      : "Y-m-d"
+
+    const theDateFormat = dateFormat === undefined ? defaultDateFormat : dateFormat
+
+    return theDateFormat
+  }
+
+  const theDateFormat = getDateFormat()
+
+  const createFlatpickrInstance = () => {
+    const options: Options = {
+      allowInput: allowInput,
+      allowInvalidPreload: allowInvalidPreload,
+      ariaDateFormat: ariaDateFormat,
+      appendTo: calendarTargetRef.current || undefined,
+      conjunction: conjunction,
+      dateFormat: theDateFormat,
+      defaultDate: (defaultDate || defaultValue) as DateOption | DateOption[],
+      defaultHour: defaultHour,
+      defaultMinute: defaultMinute,
+      disable: disable as DateLimit<DateOption>[],
+      enableSeconds: enableSeconds,
+      enableTime: enableTime,
+      hourIncrement: hourIncrement,
+      locale: (locale || undefined) as LocaleKey | Partial<CustomLocale>,
+      maxDate: (maxDate || undefined) as DateOption,
+      minDate: (minDate || undefined) as DateOption,
```

Comment:
Converting/checking for a valid date type would be nicer as there's potential to cast incorrect values to a date type and then treat it as such, without any checks.

Reply:
explained in f2f conversation that it is ok in this case